control-plane/agents/src/bin/core/controller/reconciler/volume/garbage_collector.rs

```diff
@@ -250,6 +258,53 @@ async fn disown_unused_replicas(
     GarbageCollector::squash_results(results)
 }
 
+/// Volume replicas could have been possibly expanded as part of volume resize,
+/// but the volume expansion failed overall leaving behind expanded replicas. So we
+/// bring back the replica size to what is expected by the volume spec.
+#[tracing::instrument(level = "debug", skip(context, volume), fields(volume.uuid = %volume.uuid(), request.reconcile = true))]
+async fn revert_expanded_replicas(
+    volume: &mut OperationGuardArc<VolumeSpec>,
+    context: &PollContext,
+) -> PollResult {
+    let target = context.specs().volume_target_nexus_rsc(volume.as_ref());
+    if let Some(tgt) = target {
+        let nexus_size = tgt.immutable_ref().size;
```

Comment:
immutable_ref is used for very specific cases where fields never change..
Also I suspect immutable_ref is probably no longer needed, needs to be checked.

```suggestion
        let nexus_size = tgt.as_ref().size;
```

Reply:
I don't see `as_ref` available with this type (ResourceMutex<T>). Throws error `method not found in ResourceMutex<NexusSpec>`. It looks to me here either we `lock and peek` using `tgt.lock().size` or use `immutable_ref` which is perhaps ok because the field `size` won't be modified underneath in this path. What do say would be preferred?

Reply:
Ah right this is on the ResourceMutex... So you need to take the guard or lock it, should be fine here since we hold the volume lock.

Reply:
ok, used lock to peek.